cardano-testnet/src/Testnet/Property/Util.hs

```diff
@@ -76,3 +80,18 @@ runInBackground act = void . H.evalM $ allocate (H.async act) cleanUp
 decodeEraUTxO :: (IsShelleyBasedEra era, MonadTest m) => ShelleyBasedEra era -> Aeson.Value -> m (UTxO era)
 decodeEraUTxO _ = H.jsonErrorFail . Aeson.fromJSON
 
+isBootstrapPhase
+  :: ( HasCallStack
+     , MonadIO m
+     , MonadTest m
+     , MonadCatch m
+     )
+  => String
+  -> H.ExecConfig
+  -> m Bool
+isBootstrapPhase eraName execConfig = do
+  ppJSON <-
+    execCliStdoutToJson execConfig [ eraName, "query", "protocol-parameters" ]
+  protocolParametersOut :: ProtocolParameters <- H.jsonErrorFail $ Aeson.fromJSON ppJSON
```

Comment:
You can get protocol parameters using `EpochStateView` like this: https://github.com/IntersectMBO/cardano-node/blob/master/cardano-testnet/src/Testnet/Components/Query.hs#L327 which is cheaper than calling CLI.

Reply:
removed the function